### ISSUE TITLE
control net：Images that produced stacking effects during training

### ISSUE THREAD

**Issue by YuanSnowing** [huggingface/diffusers#3765]
### Describe the bug

I tried to train control net using my own dataset, but I found the resulting images very strange. I haven't made any changes to the code (only the data input part), and the dataset I used is a 512 resolution image cut from Cityscape. But I don't think this is related to the dataset because I tried to generate images before starting training and still produced such results.
![it0_img0_1686630783](https://github.com/huggingface/diffusers/assets/53067358/ac0fa94d-f5e6-4d65-802c-e59783e7ad95)
![it0_img0_1686630882](https://github.com/huggingface/diffusers/assets/53067358/45c626d5-f8c7-436c-a8b7-60a4959ced8e)
![it0_img0_1686630931](https://github.com/huggingface/diffusers/assets/53067358/1748ec1b-cb78-4029-a078-312195b325cb)

### Reproduction

accelerate launch train_controlnet.py  \
--pretrained_model_name_or_path=$MODEL_DIR  \
--output_dir=$OUTPUT_DIR  \
--dataset_name="/data/cityscape"  \
--resolution=512  \
--learning_rate=1e-5  \
--validation_image "./1.jpg" \
--validation_prompt "a cat" \
--train_batch_size=1  \
--gradient_accumulation_steps=4  \
--gradient_checkpointing  \
--use_8bit_adam  \
--enable_xformers_memory_efficient_attention  \
--set_grads_to_none 

### System Info

- `diffusers` version: 0.17.0.dev0
- Platform: Linux-4.15.0-99-generic-x86_64-with-glibc2.10
- Python version: 3.8.0
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Huggingface_hub version: 0.14.1
- Transformers version: 4.29.1
- Accelerate version: 0.19.0
- xFormers version: 0.0.19
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by alexblattner** [comment]
you're the problem here

**Comment by YuanSnowing** [comment]
> you're the problem here

sorry, I didn't understand what you meant...where?

**Comment by alexblattner** [comment]
it isn't something that can be solved with theory alone. You have to understand the model on an instinctual level. You either, trained too much, not enough or with bad data. It's all it is in the end.

**Comment by YuanSnowing** [comment]
> 这不是仅靠理论就能解决的。你必须在本能的层面上理解模型。你要么训练太多，要么训练不足，要么数据不好。最后就这样了。

I'm not very sure.. According to the original meaning of control net, in the case of zero convolution,  inferring before starting training should be equivalent to directly using SD , so it should not be related to the training round or the quality of the training data.. Therefore, I don't know where to start to solve the problem....
but still thank you very much for your help OvO!

**Comment by williamberman** [comment]
can you post a fully reproducible setup?

**Comment by yiyixuxu** [comment]
these images.... 😱😱😱😱😱

**Comment by YuanSnowing** [comment]
> can you post a fully reproducible setup?

I download a cityscape dataset from https://huggingface.co/datasets/Chris1/cityscapes, and give it captions by blip..
processing like this

`from datasets import load_dataset, load_from_disk
from PIL import Image
import requests
import torch
from torchvision import transforms
from torchvision.transforms.functional import InterpolationMode
from models.blip import blip_decoder
import pickle
import os
import random
from datasets import Dataset
from io import StringIO,BytesIO

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
data_dic = {"image":[], "conditioning_image":[], "text":[]}
target_size = (512, 512)
crop_size = 512

def get_dataset():
    path = "/data/yxn/kitti_gen/BLIP/cityscapes/data/"
    data_list = []
    data_list.extend([path+f"train-0000{i}-of-00014.parquet" for i in range(10)])
    data_list.extend([path+f"train-000{i}-of-00014.parquet" for i in range(10,14)])
    data_list.extend([path+f"validation-0000{i}-of-00003.parquet" for i in range(3)])
    dataset = load_dataset("parquet", data_files=data_list)
    print(dataset)
    return dataset

def load_model():
    model_path = '/data/yxn/kitti_gen/BLIP/models/model_base_capfilt_large.pth'
    model = blip_decoder(pretrained=model_path, image_size=384, vit='base')
    model.eval()
    model = model.to(device)
    return model

def load_image(img,device=device):
    raw_image = img.convert('RGB')   
    w,h = raw_image.size
    transform = transforms.Compose([
        transforms.Resize((384,384),interpolation=InterpolationMode.BICUBIC),
        transforms.ToTensor(),
        transforms.Normalize((0.48145466, 0.4578275, 0.40821073), (0.26862954, 0.26130258, 0.27577711))
        ]) 
    image = transform(raw_image).unsqueeze(0).to(device)   
    return image

def main():
    dataset = get_dataset()
    model = load_model()
    counter = 0
    with torch.no_grad():
        for one in dataset["train"]:
            image1 = Image.open(BytesIO(one["image"]['bytes']))
            image2 = Image.open(BytesIO(one["semantic_segmentation"]['bytes']))
            for j in range(20):
                start_x = random.randint(0, 1524)
                start_y = random.randint(0, 500)
                image_new1 = image1.crop((start_x, start_y, start_x + crop_size, start_y + crop_size))
                image_new2 = image2.crop((start_x, start_y, start_x + crop_size, start_y + crop_size))
                image_load = load_image(image_new1,device=device)
                caption = model.generate(image_load, sample=False, num_beams=3, max_length=20, min_length=5)
                counter += 1
                print(f'{counter}: ({start_x}, {start_y}) {caption[0]}')
                data_dic["image"].append(image_new1.resize(target_size))
                data_dic["conditioning_image"].append(image_new2.resize(target_size))
                data_dic["text"].append(str(caption[0]))
    dataset_new = Dataset.from_dict(data_dic)
    dataset_new.save_to_disk("/data/cityscape")


if __name__ == "__main__":
    main()
`

then I just run the script 

`
accelerate launch train_controlnet.py  \
--pretrained_model_name_or_path=$MODEL_DIR  \
--output_dir=$OUTPUT_DIR  \
--dataset_name="/data/cityscape"  \
--resolution=512  \
--learning_rate=1e-5  \
--validation_image "./111.jpg" \
--validation_prompt "a cat" \
--train_batch_size=1  \
--gradient_accumulation_steps=4  \
--gradient_checkpointing  \
--use_8bit_adam  \
--enable_xformers_memory_efficient_attention  \
--set_grads_to_none
`
